### PR TITLE
New template interface

### PR DESCRIPTION
**Pull request by mailund**
This is the new `target_from_template` function we discussed in #151 . It is not tested--I just removed the tests we had for templates since they worked directly on `Target` objects and didn't go through the workflow interface. I'm not entirely sure how to test this now.

Please check the documentation string as well. I'm not sure how the markup there works.

### CODE REVIEW

**Comment by mailund** [comment]
I'm going to update the workflow examples now and test those.

File: gwf/core.py [hunk 1]
```diff
@@ -238,6 +230,41 @@ def target(self, name, inputs, outputs, **options):
...
+        inputs, outputs, target_options, spec = template
+
+        merged_options = self.defaults.copy()
+        merged_options.update(target_options)
+        merged_options.update(options)
+
+        new_target = Target(
+            name, inputs, outputs, merged_options,
+            working_dir=self.working_dir,
+        ) << spec
```
Comment (dansondergaard) [inline gwf/core.py]: I'd set the spec through the `spec` argument here. I reserve `<<` for "outside the core".

File: gwf/core.py [hunk 2]
```diff
@@ -238,6 +230,41 @@ def target(self, name, inputs, outputs, **options):
...
+
+        :param str name: Name of the target.
+        :param tuple template: Target specification based on a template.
+
+        Any further keyword arguments are passed to the backend and will
+        override any options provided by the template.
+        """
+        inputs, outputs, target_options, spec = template
+
+        merged_options = self.defaults.copy()
```
Comment (dansondergaard) [inline gwf/core.py]: Use `new_target.inherit_options(self.defaults)` instead.

File: gwf/core.py [hunk 3]
```diff
@@ -238,6 +230,41 @@ def target(self, name, inputs, outputs, **options):
...
+
+            workflow = Workflow()
+            workflow.target_from_template('NewTarget', my_template())
+
+        This will create a new target named `NewTarget`, configure it based
+        on the specification in the template `my_template`, and
+        add it to the workflow.
+
+        :param str name: Name of the target.
+        :param tuple template: Target specification based on a template.
```
Comment (dansondergaard) [inline gwf/core.py]: Maybe change the description to "Target specification of the form (inputs, outputs, options, spec)"?

File: gwf/core.py [hunk 4]
```diff
@@ -238,6 +230,41 @@ def target(self, name, inputs, outputs, **options):
         self._add_target(new_target)
         return new_target
 
+    def target_from_template(self, name, template, **options):
+        """Create a target and add it to the :class:`gwf.Workflow`.
+
+        This is syntactic sugar for creating a new :class:`~gwf.Target` and
+        adding it to the workflow. The target is also returned from the method
+        so that the user can directly manipulate it, if necessary. For example,
```
Comment (dansondergaard) [inline gwf/core.py]: Remove "For example, ...". It's confusing since that's not what the example is showing.

File: gwf/core.py [hunk 5]
```diff
@@ -238,6 +230,41 @@ def target(self, name, inputs, outputs, **options):
         self._add_target(new_target)
         return new_target
 
+    def target_from_template(self, name, template, **options):
+        """Create a target and add it to the :class:`gwf.Workflow`.
```
Comment (dansondergaard) [inline gwf/core.py]: Add "... from a template ..."

Comment (mailund) [inline gwf/core.py]: done

File: gwf/core.py (hunk 4 above)
Comment (mailund) [inline gwf/core.py]: done

File: gwf/core.py (hunk 2 above)
Comment (mailund) [inline gwf/core.py]: why?

File: gwf/core.py (hunk 1 above)
Comment (mailund) [inline gwf/core.py]: done

**Pull request merged**